### PR TITLE
GCG-28 Improve json format

### PR DESCRIPTION
**Pull request by DanishAlsayed**

### CODE REVIEW

File: codegen/src/test/resources/q3/Q3.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    {
+      "PrimaryRelation": "Orders",
+      "ForeignRelation": "Lineitem"
+    },
+    {
+      "PrimaryRelation": "Orders",
+      "ForeignRelation": "Customer"
+    }
+  ],
+  "LineitemProcessFunction": {
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:12]: All Relation Process Functions should represents like the join structure with an array.

File: codegen/src/test/resources/q3/Q3.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          "type": "constant",
+          "var_type": "Date",
+          "value": "1995-03-15"
+        },
+        "operator": "<"
+      }
+    }
+  },
+  "CustomerProcessFunction": {
+    "name": "Q3Orders",
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:71]: The name is incorrect.

File: codegen/src/test/resources/q3/Q3.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    "relation": "orders",
+    "this_key": [
+      "custkey"
+    ],
+    "next_key": [
+      "orderkey"
+    ],
+    "child_nodes": 1,
+    "is_Root": true,
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:51]: The relation is not root.

File: codegen/src/test/resources/q3/Q3.json [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  "CustomerProcessFunction": {
+    "name": "Q3Orders",
+    "relation": "orders",
+    "this_key": [
+      "custkey"
+    ],
+    "next_key": [
+      "custkey"
+    ],
+    "child_nodes": 1,
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:79]: Customer does not have child.

File: codegen/src/test/resources/q3/Q3.json [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    "next_key": [
+      "orderkey"
+    ],
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:56]: Try make all values into an array 'value' like the join_structure.

File: codegen/src/test/resources/q3/Q3.json [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        "operator": "="
+      }
+    }
+  },
+  "AggregateProcessFunction": {
+    "name": "Q3Aggregate",
+    "this_key": ["orderkey"],
+    "next_key": ["orderkey","orderdate","shippriority"],
+    "delta_output": true,
+    "AggregateValue": {
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:104]: I think the aggregate value is defined as SUM(l_extendedprice * (1 - l_discount)) so there should be a recursive in the expression to express the (1-discount) part.

File: codegen/src/test/resources/q3/Q3.json [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
+        "left_field": {
+          "type": "attribute",
+          "name": "orderdate"
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:59]: This one should be o_orderdate? Please check if the attribute name matches our schema in the codegen.

File: codegen/src/test/resources/q3/Q3.json [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
+        "left_field": {
+          "type": "attribute",
+          "name": "mktsegment"
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:88]: Also here should be c_mktsegment?

File: codegen/src/test/resources/q3/Q3.json [hunk 9]
```diff
@@ -0,0 +1,120 @@
+{
+  "join_structure": [
+    {
+      "PrimaryRelation": "Orders",
+      "ForeignRelation": "Lineitem"
+    },
+    {
+      "PrimaryRelation": "Orders",
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:8]: Here the primary relation is Customer and the foreign relation is Orders.

File: codegen/src/test/resources/q3/Q3.json (hunk 6 above)
Comment (DanishAlsayed) [reply codegen/src/test/resources/q3/Q3.json:104]: Can you give me an example?

File: codegen/src/test/resources/q3/Q3.json (hunk 5 above)
Comment (DanishAlsayed) [reply codegen/src/test/resources/q3/Q3.json:56]: that can only be done when there are multiple values

File: codegen/src/test/resources/q3/Q3.json [hunk 10]
```diff
@@ -5,101 +5,109 @@
...
+      "next_key": [
+        "orderkey"
+      ],
+      "child_nodes": 0,
+      "is_Root": true,
+      "is_Last": true,
+      "rename_attribute": null,
+      "select_conditions": {
+        "operator": "&&",
+        "value1": {
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:28]: The value1 can make as an array named "value", which allows arbitrary number of values.

File: codegen/src/test/resources/q3/Q3.json [hunk 11]
```diff
@@ -5,101 +5,109 @@
...
+    "LineitemProcessFunction": {
+      "name": "Q3lineitem",
+      "relation": "lineitem",
+      "this_key": [
+        "orderkey"
+      ],
+      "next_key": [
+        "orderkey"
+      ],
+      "child_nodes": 0,
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:22]: Lineitem has one child node.

File: codegen/src/test/resources/q3/Q3.json [hunk 12]
```diff
@@ -5,101 +5,109 @@
...
+    "CustomerProcessFunction": {
+      "name": "Q3Customer",
+      "relation": "orders",
+      "this_key": [
+        "custkey"
+      ],
+      "next_key": [
+        "custkey"
+      ],
+      "child_nodes": 1,
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:80]: Customer has no child.

File: codegen/src/test/resources/q3/Q3.json [hunk 13]
```diff
@@ -5,101 +5,109 @@
...
+      "is_Root": true,
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:81]: the node is not the root.

File: codegen/src/test/resources/q3/Q3.json [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+      "is_Last": true,
+      "rename_attribute": null,
+      "select_conditions": {
+        "operator": "&&",
+        "values": [
+          {
+            "left_field": {
```
Comment (wqc0712) [inline codegen/src/test/resources/q3/Q3.json:92]: Do we still need the left_field and the right_field here?

**Pull request merged**